scripts/generate_resource/main.go

```diff
@@ -0,0 +1,99 @@
+package main
+
+import (
+	"fmt"
+	"log"
+	"os"
+	"regexp"
+	"strings"
+	"text/template"
+
+	"github.com/gertd/go-pluralize"
+	"github.com/iancoleman/strcase"
+)
+
+const validResourceName = regexp.MustCompile(`^[a-zA-Z_]+$`).MatchString
+
+func generateResourceTemplate(name string) ResourceTemplate {
+	var pluralName string
+	pluralize := pluralize.NewClient()
+
+	if pluralize.IsPlural(name) {
+		pluralName = name
+		name = pluralize.Singular(name)
+	} else {
+		pluralName = pluralize.Plural(name)
+	}
+
+	return ResourceTemplate{
+		PrimaryTag:        name,
+		Name:              strings.ReplaceAll(name, "_", " "),
+		PluralName:        strings.ReplaceAll(pluralName, "_", " "),
+		Resource:          strcase.ToCamel(name),
+		ResourceInterface: strcase.ToCamel(pluralName),
+		ResourceStruct:    strcase.ToLowerCamel(pluralName),
+		ResourceID:        fmt.Sprintf("%sID", strcase.ToLowerCamel(name)),
+		ListOptions:       fmt.Sprintf("%sListOptions", strcase.ToCamel(name)),
+		ReadOptions:       fmt.Sprintf("%sReadOptions", strcase.ToCamel(name)),
+		CreateOptions:     fmt.Sprintf("%sCreateOptions", strcase.ToCamel(name)),
+		UpdateOptions:     fmt.Sprintf("%sUpdateOptions", strcase.ToCamel(name)),
+	}
+}
```

Comment:
`strcase.ToCamel(name)` , `stracse.ToCamel(pluralName)`, should probably be refactored and stored as a variable.